dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/token/TokenResponseValidator.java

```diff
@@ -190,14 +219,41 @@ private JSONObject getProviderDiscoveryMetadadata() {
 
     /**
      * @return
+     * @throws OidcDiscoveryException
      */
-    private String getJwkUri() {
-        String jwkuri = null;
-        jwkuri = clientConfig.getProviderMetadata().getJwksURI();
-        if ((jwkuri == null || jwkuri.isEmpty()) && getProviderDiscoveryMetadadata() != null) {
-            jwkuri = (String) (this.discoveredProviderMetadata.get("jwks_uri"));
+    private String getJwksUri() throws OidcDiscoveryException {
+        String jwksUri = getJwksUriFromProviderMetadata();
+        if (jwksUri != null && !jwksUri.isEmpty()) {
```

Comment:
It's a nit, but `getJwksUriFromProviderMetadata()` should already be verifying that `jwksUri` isn't empty, so you should just be able to check for `!= null` here.